cmd/nvidia-dra-controller/allocations.go

```diff
@@ -103,6 +103,9 @@ func (p *PerNodeAllocatedClaims) RemoveNode(claimUID, node string) {
 	}
 
 	delete(p.allocations[claimUID], node)
+	if len(p.allocations) == 0 {
```

Comment:
Should this be:
```suggestion
	if len(p.allocations[claimUID]) == 0 {
```
instead?

Reply:
Ah yes, sorry for the stupid, added tests for validation.